qemu/qemu.go

```diff
@@ -746,20 +761,28 @@ func (vhostuserDev VhostUserDevice) QemuParams(config *Config) []string {
 	charParams = append(charParams, fmt.Sprintf("id=%s", vhostuserDev.CharDevID))
 	charParams = append(charParams, fmt.Sprintf("path=%s", vhostuserDev.SocketPath))
 
+	switch vhostuserDev.VhostUserType {
 	// if network based vhost device:
-	if vhostuserDev.VhostUserType == VhostUserNet {
+	case VhostUserNet:
 		netParams = append(netParams, "type=vhost-user")
 		netParams = append(netParams, fmt.Sprintf("id=%s", vhostuserDev.TypeDevID))
 		netParams = append(netParams, fmt.Sprintf("chardev=%s", vhostuserDev.CharDevID))
 		netParams = append(netParams, "vhostforce")
 
 		devParams = append(devParams, VhostUserNet)
 		devParams = append(devParams, fmt.Sprintf("netdev=%s", vhostuserDev.TypeDevID))
-		devParams = append(devParams, fmt.Sprintf("mac=%s", vhostuserDev.MacAddress))
-	} else {
+		devParams = append(devParams, fmt.Sprintf("mac=%s", vhostuserDev.Address))
+	case VhostUserSCSI:
 		devParams = append(devParams, VhostUserSCSI)
 		devParams = append(devParams, fmt.Sprintf("id=%s", vhostuserDev.TypeDevID))
 		devParams = append(devParams, fmt.Sprintf("chardev=%s", vhostuserDev.CharDevID))
+	case VhostUserBlk:
+		devParams = append(devParams, VhostUserBlk)
+		devParams = append(devParams, "logical_block_size=4096")
+		devParams = append(devParams, "size=512M")
```

Comment:
block size and size should be configurable

Reply:
hey @devimc agreed for sure, I had noted that in the PR message.  -- there's a follow up PR for handling this and multi queue support.